src/foam/doc/DocBrowser.js

```diff
@@ -532,3 +559,954 @@ foam.debug.doc = function(opt_obj, showUnused) {
+    foam.u2.CSS.create({
+      code: function() {
+        /*
+               ^ { width:1200px; margin: 20px; }
+               ^ canvas { border: 1px solid black; }
+               ^ .foam-u2-ActionView- { margin: 10px; }
+               ^ input[type='range'] { width: 400px; }
+               */
+      }
+    })
+  ],
+
+  properties: [
+    'feedback_',
+    {
+      name: 'selected',
+      postSet: function(o, n) {
+        if (o) o.color = this.UNSELECTED_COLOR;
+        if (n) n.color = this.SELECTED_COLOR;
+      }
+    },
+    {
+      name: 'canvas',
+      factory: function() {
+        return this.Box.create({
+          width: 1200,
+          height: 1000,
+          color: '#f3f3f3'
+        });
+      }
+    },
+    {
+      class: 'String',
+      name: 'className',
+      value: 'className',
+    },
+    {
+      name: 'elementMap'
+    },
+    {
+      class: 'String',
+      name: 'prop',
+      value: 'propName',
+    },
+    {
+      name: 'triangleSize',
+      value: 5,
+    },
+    {
+      name: 'dashedstep',
+      value: 10,
+    },
+    'data'
+  ],
+
+  methods: [
+    function initE() {
+      var data = this.data;
+      this.className = this.data.name;
+      this.elementMap = new Map();
+
+      var widthCenterBox = 200;
+      var heightCenterBox = 30;
+
+      this.addModel(this.canvas.width / 2 - widthCenterBox / 2, this.canvas.height / 2 - heightCenterBox * 2);
+      this.addExtends(this.canvas.width / 2 - widthCenterBox / 2, this.canvas.height / 2 - heightCenterBox * 2);
+      this.addImplements(this.canvas.width / 2 - widthCenterBox / 2, this.canvas.height / 2 - heightCenterBox * 2);
+      this.addRequires(this.canvas.width / 2 - widthCenterBox / 2, this.canvas.height / 2 - heightCenterBox * 2);
+      this.addRequiredBy(this.canvas.width / 2 - widthCenterBox / 2, this.canvas.height / 2 - heightCenterBox * 2);
+      this.addSubClasses(this.canvas.width / 2 - widthCenterBox / 2, this.canvas.height / 2 - heightCenterBox * 2);
+      /*this.addImports(this.canvas.width / 2 - widthCenterBox / 2, this.canvas.height / 2 - heightCenterBox * 2);
+      this.addExports(this.canvas.width / 2 - widthCenterBox / 2, this.canvas.height / 2 - heightCenterBox * 2);*/
+      this.addRelatedto(this.canvas.width / 2 - widthCenterBox / 2, this.canvas.height / 2 - heightCenterBox * 2);
+      this.addRelatedFrom(this.canvas.width / 2 - widthCenterBox / 2, this.canvas.height / 2 - heightCenterBox * 2);
+      this.addLegend();
+
+      this.
+      addClass(this.myClass()).
+      start('center').
+      tag('br').
+      start(this.canvas).
+      on('click', this.onClick).
+      end().
+      end();
+    },
+
+    function sign(ex, sx) {
+      if (ex - sx > 0) {
+        return 1;
+      } else if (ex - sx < 0) {
+        return -1;
+      } else {
+        return 0;
+      }
+    },
+
+    function dashedLine(d, sx, sy, ex, ey) {
+      var dashedLine, changeX = 0,
```

Comment:
No need to assign to 0 if you're just reassigning on next two lines. Just set to sign() values when declaring.